chain/neutrino.go

```diff
@@ -797,16 +797,22 @@ out:
 	s.wg.Done()
 }
 
-// getNewRescanner injects the Rescanner constructor when called and defaults
+// getNewRescanFunc injects the Rescanner constructor when called and defaults
 // to using neutrino.NewRescan when unspecified.
-func (s *NeutrinoClient) getNewRescanner() rescan.NewFunc {
-	if s.newRescanner == nil {
-		s.newRescanner = func(ropts ...neutrino.RescanOption) rescan.Interface {
-			cs := &neutrino.RescanChainSource{
-				ChainService: s.CS.(*neutrino.ChainService),
-			}
-			return neutrino.NewRescan(cs, ropts...)
+func (s *NeutrinoClient) getNewRescanFunc() rescan.NewFunc {
+	if s.newRescanFunc == nil {
+		// Must cast CS as *neutrino.ChainService
+		cs := s.CS.(*neutrino.ChainService)
```

Comment:
This cast can be avoided when we instead pass in the whole function that creates a rescan (see my diff).